From 3fdb2494c1e24c0a020f5b54022d2c751fd26f50 Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Thu, 28 Nov 2013 10:52:18 +0100
Subject: login: revert lazy session-activation on non-VT seats

Existing applications like gdm already depend on new sessions to get
immediately activated on seats without VTs. Fixes a bug reported as:
  [systemd-devel] systemd 208:trouble with inactive user sessions at non-seat0 seats

This patch restores the original behavior. We either need to add a new
flag for session-creation or some other heuristic to avoid activating new
sessions in the future.

diff --git a/src/login/logind-seat.c b/src/login/logind-seat.c
index b30c4ce..ca0e8d7 100644
--- a/src/login/logind-seat.c
+++ b/src/login/logind-seat.c
@@ -413,8 +413,8 @@ int seat_attach_session(Seat *s, Session *session) {
         seat_send_changed(s, "Sessions", NULL);
 
         /* On seats with VTs, the VT logic defines which session is active. On
-         * seats without VTs, we automatically activate the first session. */
-        if (!seat_has_vts(s) && !s->active)
+         * seats without VTs, we automatically activate new sessions. */
+        if (!seat_has_vts(s))
                 seat_set_active(s, session);
 
         return 0;
-- 
cgit v0.10.2

